nessie/lib/berkeley.py

```diff
@@ -162,14 +162,11 @@ def canvas_terms():
 
 
 def future_term_ids():
-    term_ids = []
-    stop_term_id = current_term_id()
-    term_id = future_term_id()
-    while True:
-        if term_id <= stop_term_id:
-            return term_ids
-        term_ids.append(term_id)
-        term_id = previous_term_id(term_id)
+    return _collect_terms(start_term_id=future_term_id(), stop_term_id=current_term_id(), inclusive=False)
+
+
+def legacy_term_ids():
+    return _collect_terms(start_term_id=earliest_term_id(), stop_term_id=earliest_legacy_term_id())
```

Comment:
@lyttam Sorry for the delayed review - it looks like there might be an inclusive gotcha at the start as well as the end of the loop? I think this code would count Fall 2016 as both a legacy and a non-legacy term.

Reply:
@paulkerschen I guess we don't want that.

Reply:
@paulkerschen amended to fix that overlap.